client/state/products-list/reducer.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	PRODUCTS_LIST_RECEIVE,
+	PRODUCTS_LIST_REQUEST,
+	PRODUCTS_LIST_REQUEST_FAILURE,
+	DESERIALIZE,
+	SERIALIZE,
+} from 'state/action-types';
+import { productsListSchema } from './schema';
+import { isValidStateWithSchema } from 'state/utils';
+
+/**
+ * Stores the complete list of products, indexed by the product key
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function items( state = null, action ) {
```

Comment:
With @gziolo's createReducer method, we can persist just by passing in the schema as the third argument:
```javascript
import { createReducer } from 'state/utils';
export const items = createReducer( null, {
    [ PRODUCTS_LIST_RECEIVE ]: ( state, action ) => action.productsList
}, productListSchema );